pygmt/clib/conversion.py

```diff
@@ -102,7 +105,10 @@ def dataarray_to_matrix(grid):
                     dim
                 )
             )
-        region.extend([coord.min(), coord.max()])
+        if registration == "GMT_GRID_PIXEL_REG":
+            region.extend([coord.min() - coord_inc / 2, coord.max() + coord_inc / 2])
+        elif registration == "GMT_GRID_NODE_REG":
+            region.extend([coord.min(), coord.max()])
```

Comment:
In GMT, `GMT_GRID_NODE_REG=0` and `GMT_GRID_PIXEL_REG=1`. Thus, the if-else statement can be simplified as code:
```
# node_offset (i.e., registration in int) is 0 or 1
region.extend([coord.min() - coord_inc / 2 * node_offset, coord.max() + coord_inc / 2 * node_offset])
```

Reply:
Awesome, then a boolean "registration" should be the way to go, and GMT.jl uses a boolean style `reg` argument too as at https://forum.generic-mapping-tools.org/t/deciding-on-default-grid-registration-and-grid-type-for-pygmt/583/4. Will wrap then in later.